src/engine.rs

```diff
@@ -30,25 +30,61 @@ pub struct Sheet {
     pub frames: HashMap<String, Cell>,
 }
 
-#[derive(Clone, Copy)]
+#[derive(Clone, Copy, Default)]
 pub struct Point {
     pub x: i16,
     pub y: i16,
 }
 
+#[derive(Clone, Copy, Default)]
 pub struct Rect {
-    pub x: f32,
-    pub y: f32,
-    pub width: f32,
-    pub height: f32,
+    pub position: Point,
+    pub width: i16,
```

Comment:
Not sure when you made the decision to go to i16. Make sure you mention it in the book.